### PR TITLE
marisa-python: fix build

### PR DESCRIPTION
**Pull request by vanzef**
###### Motivation for this change
Fixed build procedure for the package.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/marisa/default.nix [hunk 1]
```diff
@@ -7,12 +7,14 @@ buildPythonPackage rec {
...
     owner = "s-yata";
     repo  = "marisa-trie";
-    rev   = "59e410597981475bae94d9d9eb252c1d9790dc2f";
-    sha256 = "0z4bf55np08q3cbi6gvj3cpw3zp8kf2d0jq6k74pjk066m7rapbb";
+    rev   = "8dba9850b89d7828ebf33b8ab84df2b54d31260b";
+    sha256 = "0pkp9fggk53lxlicfwrskgx33qplc4v6njbavlnz4x4z63zd4933";
   };
 
   nativeBuildInputs = [ swig marisa ];
   buildinputs = [ marisa ];
```
Comment (Mic92) [inline pkgs/development/python-modules/marisa/default.nix]: There is a typo in here: `buildInputs` and then the `NIX_CFLAGS_LINK` line should be no longer necessary.

Comment (vanzef) [reply pkgs/development/python-modules/marisa/default.nix]: You're right, `NIX_CFLAGS_COMPILE` isn't necessary either. Strange it was working with the typo until recently.

**Comment by xeji** [comment]
@GrahamcOfBorg  build python27Packages.marisa python37Packages.marisa

**Comment by Mic92** [comment]
Is this package supposed to be compatible with python3?

```
$ nix build --no-link --keep-going --max-jobs 4 --option build-use-sandbox true -f /home/joerg/.cache/nix-review/pr-57013/build.nix
error: build of '/nix/store/6bqgikqaxnbmsf9h5prnw44zsqq3awaf-python3.7-marisa-1.3.40.drv' on 'ssh://nix@eve.thalheim.io' failed: builder for '/nix/store/6bqgikqaxnbmsf9h5prnw44zsqq3awaf-python3.7-marisa-1.3.40.drv' failed with exit code 1
builder for '/nix/store/6bqgikqaxnbmsf9h5prnw44zsqq3awaf-python3.7-marisa-1.3.40.drv' failed with exit code 1; last 10 log lines:
  marisa-swig_wrap.cxx:2549:13: note: suggested alternative: ‘PyLong_FromVoidPtr’
         obj = PyCObject_FromVoidPtr(descriptor, NULL);
               ^~~~~~~~~~~~~~~~~~~~~
               PyLong_FromVoidPtr
  marisa-swig_wrap.cxx: At global scope:
  marisa-swig_wrap.cxx:2512:22: warning: ‘swig_empty_runtime_method_table’ defined but not used [-Wunused-variable]
     static PyMethodDef swig_empty_runtime_method_table[] = { {NULL, NULL, 0, NULL} };/* Sentinel */
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  error: command 'gcc' failed with exit status 1
  builder for '/nix/store/6bqgikqaxnbmsf9h5prnw44zsqq3awaf-python3.7-marisa-1.3.40.drv' failed with exit code 1
cannot build derivation '/nix/store/rb736bdh5v642fv8x9zq4f0f4fqb11q8-env.drv': 1 dependencies couldn't be built
[5 built (1 failed), 18 copied (120.4 MiB), 31.1 MiB DL]
error: build of '/nix/store/rb736bdh5v642fv8x9zq4f0f4fqb11q8-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/57013
1 package failed to build:
python37Packages.marisa
```

Otherwise please use the `disable` attribute to mark it as broken on python3.

**Comment by vanzef** [comment]
> Is this package supposed to be compatible with python3?

I guess it isn't since `CObject API` is considered deprecated since Python 3.1 and the only package (AFAIK)  which depends on `pythonPackages.marisa` -- is `libkkc` which uses `python2Packages`.

> Otherwise please use the disable attribute to mark it as broken on python3.

Done.

**Comment by xeji** [comment]
@GrahamcOfBorg build python27Packages.marisa

**Comment by xeji** [comment]
nix-review looks ok now

**Comment by xeji** [comment]
backported to 19.03: 52e59ba4ad1 , cc #56826

**Pull request merged**